build-external-api-jar/tests/src/test/scala/org/virtuslab/tests/OpenProjectAndBuildArtifacts.scala

```diff
@@ -0,0 +1,166 @@
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.rmic" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.scripting.nashorn" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.scripting.nashorn.shell" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.sctp" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.security.auth" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.security.jgss" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.unsupported" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.unsupported.desktop" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.xml.dom" type="simple" />
+      |            <root url="jrt:///usr/lib/jvm/java-11-openjdk-amd64!/jdk.zipfs" type="simple" />
+      |          </root>
+      |        </classPath>
+      |        <javadocPath>
+      |          <root type="composite">
+      |            <root url="https://docs.oracle.com/en/java/javase/11/docs/api/" type="simple" />
+      |          </root>
+      |        </javadocPath>
+      |        <sourcePath>
+      |          <root type="composite" />
+      |        </sourcePath>
+      |      </roots>
+      |      <additional />
+      |    </jdk>
+      |  </component>
+      |</application>
+      """.stripMargin
+  }
+   val artifactVal =
+      """
+        |<component name="ArtifactManager">
+        |  <artifact type="jar" name="external-system-test-api">
+        |    <output-path>$PROJECT_DIR$/out/artifacts/external_system_test_api</output-path>
+        |    <root id="archive" name="external-system-test-api.jar">
+        |      <element id="module-test-output" name="intellij.platform.externalSystem.impl" />
+        |      <element id="module-test-output" name="intellij.platform.externalSystem.tests" />
+        |    </root>
+        |  </artifact>
+        |
+        |
+        |</component>""".stripMargin
+}
+
+
+class OpenProjectAndBuildArtifacts extends IdeProbeTestSuite with ProbeExtensions {
+
+  import OpenProjectAndBuildArtifacts._
+
+  registerFixtureTransformer(
+    _.withAfterIntelliJInstall{ (_, installedIntelliJ) =>
+      installedIntelliJ.paths.config.resolve( "options/jdk.table.xml" ).write(OpenProjectAndBuildArtifacts.jdkDef)
+    }
+  )
+  registerFixtureTransformer(
+    _.withAfterWorkspaceSetup{(_, path) =>
+      path.resolve(".idea/artifacts/external_system_test_api.xml").write(artifactVal)
+    }
+  )
+
+  @Test def build: Unit = fixtureFromConfig().run { intelliJ =>
+    val project = intelliJ.probe.openProject(intelliJ.workspace)
+    Thread.sleep(10*1000)
+
+    def waitUntilTrueForAllTasks(predicate: Seq[String] => Boolean) = {
```

Comment:
Consider synchronizing this method just in case.

Reply:
Done here https://github.com/pantsbuild/intellij-pants-plugin/pull/599